toxcore/crypto_core.h

```diff
@@ -230,6 +263,27 @@ void increment_nonce_number(uint8_t *nonce, uint32_t increment);
  */
 void new_symmetric_key(uint8_t *key);
 
+#ifndef VANILLA_NACL
```

Comment:
The `#ifdef` is probably not needed. Also not done for `create_extended_keypair`. I think we can use this function freely throughout the codebase, but in case of `VANILLA_NACL`, this function simply does nothing (slightly less secure than with sodium, but ~nobody uses it anyway).